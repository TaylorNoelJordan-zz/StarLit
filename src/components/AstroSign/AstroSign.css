.astro-symbol {
    background-color: #9cd8cd;
    border-radius: 30px;
    height: 100px;
    opacity: 0.8;
    padding: 8px;
}