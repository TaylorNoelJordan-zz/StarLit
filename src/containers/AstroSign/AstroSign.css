.astro-symbol {
    background-color: #9cd8cd;
    border-radius: 30px;
    height: 100px;
    margin: 15px;
    opacity: 0.8;
    padding: 8px;
    width: 100px;
}

.date-range {
    color: #9cd8cd;
}