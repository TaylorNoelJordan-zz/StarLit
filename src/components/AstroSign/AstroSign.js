import React from 'react';
import './AstroSign.css'

const AstroSign = () => {
    return (
        <img src='' alt='astro sign symbol' className='astro-symbol'/>
    )
}

export default AstroSign;