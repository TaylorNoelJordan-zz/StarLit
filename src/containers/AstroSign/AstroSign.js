import React from 'react';
import './AstroSign.css'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const AstroSign = (props) => {
    const { sign, dateRange } = props
    const sun = 'sun'
    return (
        <>
        <img src={require(`../../../signImages/${sign || sun}.svg`)} alt='astro sign symbol' className='astro-symbol'/>
        <p className='date-range'>{dateRange}</p>
        </>

    )
}

export const mapStateToProps = state => ({
    sign: state.user.sign,
    dateRange: state.user.date_range
})

AstroSign.propTypes = {
    sign: PropTypes.string,
    dateRange: PropTypes.string
}

export default connect(mapStateToProps)(AstroSign);