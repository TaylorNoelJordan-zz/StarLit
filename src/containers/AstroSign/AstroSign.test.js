import React from 'react';
import { AstroSign, mapStateToProps } from './AstroSign';
import { shallow } from 'enzyme';

describe('AstroSign', () => {
    let wrapper;
    

    beforeEach(() => {
        wrapper = shallow(<AstroSign/>)
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('mapStateToProps', () => {
        it('should return the corresponding portions of state', () => {
            const mockState = {
                user : {
                    sign: 'Virgo',
                    date_range: 'Aug 23 - Sep 22'
                }
            }

            const expected = {
                sign: mockState.user.sign,
                dateRange: mockState.user.date_range
            }

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected)

        })
    })
})