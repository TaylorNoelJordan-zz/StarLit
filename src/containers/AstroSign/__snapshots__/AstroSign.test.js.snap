// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstroSign should match the snapshot 1`] = `ShallowWrapper {}`;
